app/assets/javascripts/tribo_viva/checkouts/checkout.js.coffee

```diff
@@ -0,0 +1,97 @@
+TriboViva.Checkouts ?= {}
+
+TriboViva.Checkouts.Checkout =
+  init: ->
+    Iugu.setAccountID("79d7660c-6169-4e07-aed9-8a0883d5249a")
+    Iugu.setTestMode(true) # disable on production
+
+    # Set mask for card expiration date
+    $(".js-credit-card-number").mask("9999 9999 9999 9999")
+    $(".js-credit-card-card-expiration").mask("99/99")
+
+    # To hide or show credit card form
+    creditCardForm          = $('#js-creditcard-form')
+    creditCardTerms         = $('.js-terms')
+    bankSlipCheckBox        = $('#js-payment-method-bank-slip')
+    creditCardCheckBox      = $('#js-payment-method-credit-card')
+    bankSlipCheckBoxLabel   = $('#js-bank-slip-checkbox-label')
+    creditCardCheckBoxLabel = $('#js-credit-card-checkbox-label')
+
+    creditCardCheckBoxLabel.on 'click', ->
+      creditCardCheckBoxLabel.addClass('active')
+      creditCardCheckBox.prop('checked', true).attr('checked', true)
+
+      bankSlipCheckBoxLabel.removeClass('active')
+      bankSlipCheckBox.prop('checked', false).attr('checked', false)
+
+      creditCardForm.show()
+
+    bankSlipCheckBoxLabel.on 'click', ->
+      bankSlipCheckBoxLabel.addClass('active')
+      bankSlipCheckBox.prop('checked', true).attr('checked', true)
+
+      creditCardCheckBoxLabel.removeClass('active')
+      creditCardCheckBox.prop('checked', false).attr('checked', false)
+
+      creditCardForm.hide()
+
+    creditCardTerms.on 'change', ->
+      if $(this).is(':checked')
+        $('.js-submit-payment').attr('disabled', false)
+      else
+        $('.js-submit-payment').attr('disabled', true)
+
+    # start form validation and submit
+    $('#js-payment-form').submit (e) ->
+      form            = $(this)
+      cc              = $('.js-credit-card-number').val()
+      cvv             = $('.js-credit-card-cvv').val()
+      brand           = Iugu.utils.getBrandByCreditCardNumber(cc)
+      cardName        = $('.js-credit-card-card-name').val()
+      cardExpiration  = $('.js-credit-card-card-expiration').val()
+
+      if creditCardCheckBox.attr('checked')
+        # first of all, handle errors on credit card form
+        if !Iugu.utils.validateCreditCardNumber(cc, brand)
```

Comment:
You think that is better to validate every time it changes? @filipelinhares what you say? Validate on form submit or when fields change?